common/server/statshouse/statshouse-client.cpp

```diff
@@ -0,0 +1,65 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "common/resolver.h"
+#include "net/net-connections.h"
+#include "runtime/critical_section.h"
+#include "statshouse-client.h"
+
+namespace statshouse {
+
+void Client::set_port(int value) {
+  this->port = value;
+}
+
+void Client::set_host(const char *value) {
+  this->host = value;
+}
+
+bool Client::init_connection() {
+  sock_fd = socket(AF_INET, SOCK_DGRAM, 0);
+  if (sock_fd < 0) {
+    vkprintf(2, "Can't create statshouse socket");
+    return false;
+  }
+  fcntl(sock_fd, F_SETFL, O_NONBLOCK);
+
+  const char *hostname = host ?: "localhost";
+  hostent *h;
+  if (!(h = gethostbyname(hostname)) || h->h_addrtype != AF_INET || h->h_length != 4 || !h->h_addr_list || !h->h_addr) {
+    vkprintf(2, "Can't resolve statshouse host\n");
+    return -1;
+  }
+
+  struct sockaddr_in addr {};
+  memset(&addr, 0, sizeof(addr));
+  addr.sin_family = AF_INET;
+  addr.sin_port = htons(port);
+  addr.sin_addr.s_addr = (*(uint32_t *)h->h_addr);
+  if (connect(sock_fd, reinterpret_cast<sockaddr *>(&addr), sizeof(addr)) < 0) {
+    vkprintf(2, "Can't connect to statshouse host: %s\n", host);
+    return false;
+  }
+  return true;
+}
+
+void Client::send_metrics(const std::vector<Metric> &metrics) {
+  if (port == 0 || (sock_fd <= 0 && !init_connection())) {
+    return;
+  }
+  tl_query_payload tl_payload = make_tl_request_payload(metrics);
+
+  send(sock_fd, reinterpret_cast<char *>(tl_payload.buffer), tl_payload.len * 4, 0);
```

Comment:
let's handle errors here (distinguish EWOULDBLOCK and other errors) and log it via `log_server_error`

Reply:
done